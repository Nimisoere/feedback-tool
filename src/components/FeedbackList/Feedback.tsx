import React from 'react'
import { FeedbackResponse } from '../../hooks/api/useGetFeedBack'
import { range } from '../../utils/common'

type Props = {
  feedback: FeedbackResponse
}

const Feedback = React.memo(function Feedback({ feedback }: Props) {
  const datestring = new Date(feedback.createdAt)?.toLocaleString()
  return (
    <article key={feedback.id} className='flex flex-col mb-4'>
      <div className="font-medium text-lg mb-2 text-black">
        <p>{feedback.name} <span className='text-slate-600 text-sm'>({feedback.email})</span></p>
      </div>
      <div className="flex items-center mb-2">
        {range(1, 5, 1).map((star) => <span className={`text-2xl ${star <= feedback.rating ? "text-yellow-400" : "text-slate-400"}`} key={star}>&#11089;</span>)}
        <p className="ml-2 text-sm font-medium text-gray-900">{feedback.rating} out of 5</p>
      </div>
      <footer className="mb-2 text-sm text-gray-500"><p>Reviewed on <time dateTime={datestring}>{datestring}</time></p></footer>
      <div className="mb-2 font-light text-gray-500">{feedback.comment}</div>
    </article>
  )
})

export default Feedback