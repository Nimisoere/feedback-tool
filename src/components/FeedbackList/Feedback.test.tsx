import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import Feedback from "./Feedback";
import { mockfeedback } from "../../utils/test/mocks";

describe("Feedback Graph", () => {
  it("should match snapshot", async () => {
    const { getByText, container } = render(
      <Feedback feedback={mockfeedback[0]} />
    );
    expect(container).toMatchSnapshot();
  });
});
