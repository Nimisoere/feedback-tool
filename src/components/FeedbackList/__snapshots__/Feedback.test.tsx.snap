// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Graph should match snapshot 1`] = `
<div>
  <article
    class="flex flex-col mb-4"
  >
    <div
      class="font-medium text-lg mb-2 text-black"
    >
      <p>
        Nimisoere
         
        <span
          class="text-slate-600 text-sm"
        >
          (
          lawson.nimi@yahoo.com
          )
        </span>
      </p>
    </div>
    <div
      class="flex items-center mb-2"
    >
      <span
        class="text-2xl text-yellow-400"
      >
        ⭑
      </span>
      <span
        class="text-2xl text-slate-400"
      >
        ⭑
      </span>
      <span
        class="text-2xl text-slate-400"
      >
        ⭑
      </span>
      <span
        class="text-2xl text-slate-400"
      >
        ⭑
      </span>
      <span
        class="text-2xl text-slate-400"
      >
        ⭑
      </span>
      <p
        class="ml-2 text-sm font-medium text-gray-900"
      >
        1
         out of 5
      </p>
    </div>
    <footer
      class="mb-2 text-sm text-gray-500"
    >
      <p>
        Reviewed on 
        <time
          datetime="19/12/2022, 10:48:11"
        >
          19/12/2022, 10:48:11
        </time>
      </p>
    </footer>
    <div
      class="mb-2 font-light text-gray-500"
    >
      lorem
    </div>
  </article>
</div>
`;
